sync/src/chain.rs

```diff
@@ -2082,16 +2082,12 @@ impl ChainSync {
 
 		if !enacted.is_empty() {
 			// Select random peers to re-broadcast transactions to.
-			let mut random = random::new();
-			let len = self.peers.len();
-			let peers = random.gen_range(0, min(len, 3));
-			trace!(target: "sync", "Re-broadcasting transactions to {} random peers.", peers);
-
-			for _ in 0..peers {
-				let peer = random.gen_range(0, len);
-				self.peers.values_mut().nth(peer).map(|mut peer_info| {
+			let peers: Vec<PeerId> = ChainSync::select_random_peers(&self.peers.keys().cloned().collect::<Vec<_>>());
+			trace!(target: "sync", "Re-broadcasting transactions to random peers.");
+			for peer in peers.iter().take(3) {
```

Comment:
I'm good with deterministic behaviour, just think that 3 peers at block might be a little bit too much (transactions will be rebroadcasted too often), that's why I suggested to send to only 1 peer - in such case we won't have two peers selected twice hence we can get rid of the allocation.